### ISSUE TITLE
Tests require pthread

### ISSUE THREAD

**Issue by QuLogic** [zeromq/cppzmq#284]
In `tests/monitor.cpp`, `std::thread` is used; it seems that g++/libstdc++'s implementation requires linking with pthreads somehow:
```
[ 58%] Linking CXX executable ../bin/unit_tests
cd .../cppzmq/build/tests && /usr/bin/cmake -E cmake_link_script CMakeFiles/unit_tests.dir/link.txt --verbose=1
/usr/lib64/ccache/c++    -rdynamic CMakeFiles/unit_tests.dir/message.cpp.o CMakeFiles/unit_tests.dir/context.cpp.o CMakeFiles/unit_tests.dir/socket.cpp.o CMakeFiles/unit_tests.dir/poller.cpp.o CMakeFiles/unit_tests.dir/active_poller.cpp.o CMakeFiles/unit_tests.dir/multipart.cpp.o CMakeFiles/unit_tests.dir/monitor.cpp.o  -o ../bin/unit_tests /usr/lib64/libzmq.so 
/usr/bin/ld: CMakeFiles/unit_tests.dir/monitor.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: //usr/lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
I have not been able to find a proper reference for this or whether it's a bug somewhere else.

**Comment by sigiesec** [comment]
This problem doesn't arise in the existing builds on travis-ci. In order to reproduce this, it would be good to know which toolchain you are using. Maybe we can add a travis-ci build to reproduce this, to ensure it won't break again. Could you attach the CMakeOutput.log

**Comment by kurdybacha** [comment]
HI @QuLogic, fix for this on its way in #285

**Comment by QuLogic** [comment]
Fixed by #285 (you should mention this in the commit message or PR description for auto-close.)
